main.cpp

```diff
@@ -188,214 +286,313 @@ struct Range
+    }
+    else
+    {
+    std::cout << "Range In Good Shape" << std::endl;
+    }
+    
+}
+bool Range::RangeControls::selfCleans (bool settingsOn)
+{
+    while(settingsOn) return settingsOn;
+    return !settingsOn;
+}
+
 struct ShoppingCart 
 {
-    // 1) its material (std::string)
     std::string cartMaterial = "aluminum";
-    // 2) number of wheels (int)
     int numWheels = 4;
-    // 3) color of its handle (std::string)
     std::string colorHandle = "red";
-    // 4) its width (float)
     float cartWidth = 24.1f;
-    // 5) which store it belongs to (std::string)
     std::string cartStore = "Albertsons";
 
-    // 1) carries groceries 
-    void carryGrocery (bool badWheels = false, float weightLimit =  40.2f);
-    // 2) it can roll down a hill
-    void rollsDownHill (bool steepHill, float velocity = 30.2f);
-    // 3) can be parked in a cart corral 
-    bool isParked (std::string location = "corral", bool moves = false);
+    bool carryGrocery (bool badWheels = false, float weightLimit =  40.2f);
+    bool rollsDownHill (bool steepHill, bool fast);
+    bool isParked (std::string cartLocation = "corral");
 };
 
-/*
-Thing 4) wind
- */
+bool ShoppingCart::carryGrocery (bool badWheels, float weight)
+{
+    if(badWheels && weight <= 40.2f) 
+        return true;
+    return false;
+}
+bool ShoppingCart::rollsDownHill (bool steepHill, bool fast)
+{
+    if(steepHill) 
+        return fast;
+    return !fast;
+}
+bool ShoppingCart::isParked (std::string cartLocation)
+{
+    if(cartLocation == "corral") 
+        return true;
+    return false;
+}
+
 struct Wind
 {
-    // 1) speed (float)
     double speed = 3.4234898;
-    // 2) direction (std::string)
-    std::string direction = "North";
-    // 3) its definition (std:: string)
+    std::string direction = "North West";
     std::string definition = "movement in air molecules";
-    // 4) its cause (std:: string)
     std::string cause = "difference in gas density";
-    // 5) its use (std:: string)
     std::string use = "wind energy";
 
-    // 1) it transports seeds
-     void transportSeeds (int numSeed = 4, float gust = 3.42f);
-    // 2) it turns wind turbines 
-     void turnTurbines (std::string direction, bool turnTurbine = true, float gust = 34.2f);
-    // 3) it helps birds travel 
-     void helpBirdTravel (std::string direction, bool birdFly, float gust = 34.2f);
-
+     bool transportSeeds (int numSeed = 4, float gust = 3.42f);
+     bool turnTurbines (std::string, float gust = 34.2f);
+     void helpBirdTravel (std::string direction2, bool birdFly, float gust = 34.2f);
 }; 
+bool Wind::transportSeeds (int numSeed, float gust)
+{
+    if(numSeed<=10 && gust > 30)
+        return true;
+    return false;
+
+}
+bool Wind::turnTurbines (std::string direction1, float gust)
+{
+    if(direction1 == "North West" && gust > 80.f)  
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise